### DISCUSSION TITLE
Request software data information

### DISCUSSION THREAD

**Discussion by tobiaslempen** [firedrakeproject/firedrake#2981] [Firedrake support]
Dear community of Firedrake ,
I am a mechanical engineering student at RPTU and currently writing my master thesis in the field of topology optimization. 
Within the scope of the work, I would like to mention detailed information about the software “firedrake” and therefore kindly ask you to support the following points:

•	Does the software provide special geometry features like mesh repair, parametric modelling, smoothing, wireframe modelling or of another kind?
•	Are there any Reverse-Engineering capabilities like point cloud import, mesh processing tools, creation of parametric models or of another kind?
•	Is some sort of consideration for additive manufacturing provided? E.g. Lattice, overhang(angle) constraint, part orientation, support structure generation, feature size constraint...
•	Is there interpretation as “feature recognition, form detection, profile-recognition, symmetry recognition, freeform surface recognition or of another kind?
•	Is the basic modelling technology Bezier, NURBS, B-Rep, CSG or of another kind?

I would really appreciate your support and look forward to your reply. Many thanks in advance.

Kind regards,
Cyrill

**Comment by colinjcotter** [comment]
Dear Cyrill,
  Firedrake is built on the assumption that you use some other independent mesh generation software and then import the mesh (with the exception of some built in mesh routines providing simple meshes for development).

all the best
--cjc

________________________________________
From: tobiaslempen ***@***.***>
Sent: 13 June 2023 06:46
To: firedrakeproject/firedrake
Cc: Subscribed
Subject: [firedrakeproject/firedrake] Request software data information (Discussion #2981)

Dear community of Firedrake ,
I am a mechanical engineering student at RPTU and currently writing my master thesis in the field of topology optimization.
Within the scope of the work, I would like to mention detailed information about the software “firedrake” and therefore kindly ask you to support the following points:

• Does the software provide special geometry features like mesh repair, parametric modelling, smoothing, wireframe modelling or of another kind?
• Are there any Reverse-Engineering capabilities like point cloud import, mesh processing tools, creation of parametric models or of another kind?
• Is some sort of consideration for additive manufacturing provided? E.g. Lattice, overhang(angle) constraint, part orientation, support structure generation, feature size constraint...
• Is there interpretation as “feature recognition, form detection, profile-recognition, symmetry recognition, freeform surface recognition or of another kind?
• Is the basic modelling technology Bezier, NURBS, B-Rep, CSG or of another kind?

I would really appreciate your support and look forward to your reply. Many thanks in advance.

Kind regards,
Cyrill
